host/containerinit/init.go

```diff
@@ -395,6 +401,71 @@ func getCmdPath(c *Config) (string, error) {
 	return cmdPath, nil
 }
 
+func monitor(port host.Port, env map[string]string) (discoverd.Heartbeater, error) {
+	config := port.Service
+	client := discoverd.NewClientWithURL(env["DISCOVERD"])
+
+	if config.Create {
+		// TODO: maybe reuse maybeAddService() from the client
+		if err := client.AddService(config.Name); err != nil {
+			if je, ok := err.(hh.JSONError); !ok || je.Code != hh.ObjectExistsError {
+				return nil, fmt.Errorf("something went wrong with discoverd: %s", err)
+			}
+		}
+	}
+	portStr := strconv.Itoa(port.Port)
+	if portStr == "0" {
+		portStr = env["PORT"]
+	}
+	localAddr := fmt.Sprintf("127.0.0.1:%s", portStr)
+	addr := fmt.Sprintf("%s:%s", env["EXTERNAL_IP"], portStr)
+	var check health.Check
+	switch config.Check.Type {
```

Comment:
Should probably add support for registering a service without a checker? Would skip the `health.Registration` and instead just return a heartbeater which gets closed when the process exits.